corehq/apps/es/README.rst

```diff
@@ -1,6 +1,174 @@
 Elasticsearch App
 =================
 
+
+Elasticsearch Index Management
+------------------------------
+
+CommCare HQ data in Elasticsearch is integral to core application functionality.
+The level that the application relies on Elasticsearch data varies from index to
+index. Currently, Elasticsearch contains both authoritative data (for example
+``@indexed_on`` case property and ``UnknownUser`` user records) and data used
+for real-time application logic (the ``users`` index, for example).
+
+In order to guarantee stability (or "manageability", if you will) of this core
+data, it is important that Elasticsearch indexes are maintained in a consistent
+state across all environments as a concrete design feature of CommCare HQ. This
+design constraint is accomplished by managing Elasticsearch index modifications
+(for example: creating indexes, updating index mappings, etc) exclusively
+through Django's migration framework. This ensures that all Elasticsearch index
+modifications will be part of standard CommCare HQ code deployment procedures,
+thereby preventing Elasticsearch index state drift between maintained CommCare
+HQ deployments.
```

Comment:
Would it be worth adding a paragraph here to the effect of

> One or more migrations are required any time index mappings, tuning parameters, ... are changed in code.

That list should be expanded to include all things in code that describe the state of elastic entities where the code change would not immediately take effect on deployment without a corresponding migration being applied.

Reply:
See fe7b237e490